e2e-tests/playwright.config.ts

```diff
@@ -1,54 +1,129 @@
 import { defineConfig, devices } from '@playwright/test'
+import { config } from 'dotenv'
+
+const CI = !!process.env.CI
+const { SLOW_MO, OURHEALTH_PARTICIPANT_URL } = process.env
 
 /**
- * Read environment variables from file.
- * https://github.com/motdotla/dotenv
+ * Pass in `TEST_ENV` environment variable to read environment variables from .env.* file. Example: TEST_ENV=local
+ *  - override hardcoded URL in global-setup.ts
+ *  - non-default environment variables
+ *  - specify new environment variables
  */
-// require('dotenv').config();
+const { TEST_ENV } = process.env
+if (TEST_ENV) {
+  config({ path: `.env.${TEST_ENV}`, override: true })
+} else {
+  config()
+}
 
 /**
  * See https://playwright.dev/docs/test-configuration.
  */
 export default defineConfig({
-  testDir: './tests',
+  testMatch: '**/*.test.ts',
+  /* timeout for each test, 30 seconds by default. */
+  timeout: 60 * 1000,
   /* Run tests in files in parallel */
   fullyParallel: true,
   /* Fail the build on CI if you accidentally left test.only in the source code. */
-  forbidOnly: !!process.env.CI,
+  forbidOnly: CI,
   /* Retry on CI only */
-  retries: process.env.CI ? 2 : 0,
+  retries: CI ? 2 : 0,
   /* Opt out of parallel tests on CI. */
-  workers: process.env.CI ? 1 : undefined,
+  workers: CI ? 1 : undefined,
   /* Reporter to use. See https://playwright.dev/docs/test-reporters */
-  reporter: 'html',
+  reporter: [
+    ['list'],
+    ['html', {  open: 'never' }]
+  ],
   /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
   use: {
     /* Base URL to use in actions like `await page.goto('/')`. */
     // baseURL: 'http://127.0.0.1:3000',
 
+    actionTimeout: 10 * 1000,
+
     /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
-    trace: 'on-first-retry'
+    trace: CI ? 'on-first-retry' : 'retain-on-failure',
+    /* https://playwright.dev/docs/videos#record-video */
+    video: CI ? 'on-first-retry' : 'on',
+
+    launchOptions: {
+      slowMo: SLOW_MO ? parseInt(SLOW_MO) : 100,
+      // Account for minor difference in text rendering and resolution between headless and headed mode
+      ignoreDefaultArgs: ['--hide-scrollbars']
+    }
   },
   expect: {
     /* faster timeouts of local tests helps speed up creating new tests, at the cost of flakiness */
-    timeout: process.env.CI ? 10000 : 5000
+    timeout: CI ? 10000 : 5000,
+    toHaveScreenshot: {
+      // Account for minor difference in text rendering and resolution between headless and headed mode
+      threshold: 1,
+      maxDiffPixelRatio: 1
+    }
   },
 
   globalSetup: require.resolve('./global-setup.ts'),
 
   /* Configure projects for major browsers */
   projects: [
+    /*  Running only the OurHealth study tests: npx playwright test --project="OurHealth-Chrome"  */
     {
-      name: 'chromium',
+      name: 'OurHealth-Chrome',
```

Comment:
Playwright [projects](https://playwright.dev/docs/test-projects).

Grouping of E2E tests based on a specific study name.